### PR TITLE
Initial commit - Add README and skeleton app with healthcheck endpoint

### PR DESCRIPTION
**Pull request by dalebradley**
Initial commit. 

Add skeleton API app, with healthcheck endpoint and README

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	Database   string `env:"MONGODB_DATABASE"           flag:"mongodb-database"           flagDesc:"MongoDB database for data"`
+	MongoDBURL string `env:"MONGODB_URL"                flag:"mongodb-url"                flagDesc:"MongoDB server URL"`
+	APIKey     string `env:"PAYMENTS_API_KEY"           flag:"api-key"                    flagDesc:"API key used to authenticate for internal API calls"`
+}
+
+// DefaultConfig returns a pointer to a Config instance that has been populated
+// with default values.
+func DefaultConfig() *Config {
+	return &Config{
+		BindAddr:   ":4040",
```
Comment (jsimons1989) [inline config/config.go]: We've had problems with hardcoded bind ports. Config has got out of sync causing pain when trying to debug (it fails within the router which confuses people).

File: main.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/companieshouse/payments.api.ch.gov.uk/config"
+	"github.com/companieshouse/payments.api.ch.gov.uk/handlers/private"
+	"github.com/companieshouse/payments.api.ch.gov.uk/handlers/public"
+
+	"github.com/gorilla/pat"
+	"github.com/justinas/alice"
+)
+
+func main() {
+	log.Namespace = "payments.api.ch.gov.uk"
```
Comment (ChrisJBurns) [inline main.go]: probably worth putting this as a global var so all packages can access. or even put it in config

File: main.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+	cfg := config.Get()
+
+	router := pat.New()
+	chain := alice.New()
+
+	public.Register(router)
+	private.Register(router)
+
+	log.Info("Starting payments.api.ch.gov.uk service")
```
Comment (ChrisJBurns) [inline main.go]: then possibly using that namespace here

File: main.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	err := http.ListenAndServe(cfg.BindAddr, chain.Then(router))
+
+	if err != nil {
+		log.Error(err)
+	}
+	log.Trace("Exiting payments.api.ch.gov.uk service")
```
Comment (ChrisJBurns) [inline main.go]: and here?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Companies House Payments API Service
+The Companies House API Service for handling payments.
```
Comment (jsimons1989) [inline README.md]: Can you add the configs to the README please. See the example here: https://github.com/companieshouse/styleguides/blob/master/READMEs.md

File: main.go (hunk 2 above)
Comment (dalebradley) [reply main.go]: Done

File: main.go (hunk 3 above)
Comment (dalebradley) [reply main.go]: Done

File: main.go (hunk 4 above)
Comment (dalebradley) [reply main.go]: Done

File: README.md (hunk 5 above)
Comment (dalebradley) [reply README.md]: Done

File: config/config.go (hunk 1 above)
Comment (dalebradley) [reply config/config.go]: Done

File: main.go (hunk 2 above)
Comment (dalebradley) [reply main.go]: Unit test's will be coming after this PR. Along with start script and make file.

File: config/config.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	MongoDBURL string `env:"MONGODB_URL"                flag:"mongodb-url"                flagDesc:"MongoDB server URL"`
+	APIKey     string `env:"PAYMENTS_API_KEY"           flag:"api-key"                    flagDesc:"API key used to authenticate for internal API calls"`
+}
+
+// DefaultConfig returns a pointer to a Config instance that has been populated
+// with default values.
+func DefaultConfig() *Config {
+	return &Config{
+		EricAddr:   ":4001",
+		Database:   "transactions",
```
Comment (macreman) [inline config/config.go]: default database should be a new one specific to payment data, not the existing transactions DB

**Pull request merged**